packages/api/endpoints/executions.js

```diff
@@ -79,6 +86,34 @@ async function del(req, res) {
   return res.send({ message: 'Record deleted' });
 }
 
+/**
+ * Get execution history for a single granule or multiple granules
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function searchByGranules(req, res) {
+  const payload = req.body;
+  const knex = await getKnexClient();
+  const granules = await getGranulesForPayload(payload, knex);
+  const { page = 1, limit = 1 } = req.query;
+  const offset = page < 1 ? 0 : (page - 1) * limit;
```

Comment:
Is `page` ever less than 1? It defaults to 1 so it's only less than 1 if someone passes in `page=0` as a query parameter I guess. 

Also do these default values mean that by default 1 result is returned per page? It seems like we could up that to 10 or so without compromising performance and require less pagination from the front-end client. Of course I guess the front-end can/will just specify these query parameters

Regardless, I think we need some unit tests to ensure that sending the query params to the endpoint makes the pagination work as expected. I know we already have unit tests to prove that the PG model handles the pagination, but our logic in this endpoint handler to pass on the pagination values needs to be validated.

Reply:
Yeah basically just wanted to have some protection in case the user passes in 0 or something else funky for the page value.

Set the default to 1 to be consistent with other endpoints (or at least what the docs say here about page and limit https://nasa.github.io/cumulus-api/) but can change if wanted

Couldn't figure out how to link to the particular line, but there is a unit for the paging here https://github.com/nasa/cumulus/pull/2324/files#diff-b8a520e8e7976a793a61b3a1df0386af448d81490730b9ec430419e655ef5d53R456 - L456 in api/tests/endpoints/test-executions.js "POST /executions/search-by-granules supports paging". Let me know if I should add any more.

Reply:
That all makes sense. Sorry I somehow missed those unit tests on the first pass